src/components/editor/editor-preview.tsx

```diff
@@ -0,0 +1,179 @@
+import PageUtils from "../../utils/page-utils";
+import { useEffect, useState } from "react";
+import { IconContainer } from "./new-page-button";
+import strings from "../../localization/strings";
+import AddQuestionDialog from "./add-question-dialog";
+import { IframeClickEvent } from "../../types";
+
+/**
+ * Styled navigation button
+ */
+export const MenuButton = styled(IconButton, {
+  label: "menu-button"
+})(({ theme }) => ({
+  width: 100,
+  height: 100,
+  position: "absolute",
+  top: theme.spacing(2),
+  right: theme.spacing(5),
+  zIndex: 100000,
+  padding: 0,
+  margin: 0
+}));
+
+/**
+ * Add Question Container props
+ */
+interface AddQuestionContainerProps {
+  borderWidth: number;
+  width: number;
+  height: number;
+  gap: number;
+}
+
+/**
+ * Styled add question container component
+ */
+const AddQuestionContainer = styled(Stack, {
+  label: "add-question-container"
+})<AddQuestionContainerProps>(({ borderWidth, width, height, gap }) => ({
+  gap: gap,
+  textAlign: "center",
+  position: "absolute",
+  bottom: "10%",
+  right: "50%",
+  transform: "translateX(50%)",
+  backgroundColor: "#007f34",
+  borderWidth: borderWidth,
+  borderStyle: "dashed",
+  borderColor: "#fff",
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+  justifyContent: "center",
+  width: width,
+  height: height
+}));
+
+/**
+ * Components properties
+ */
+interface Props {
+  htmlString: string;
+  width: number;
+  height: number;
+  scale: number;
+  page: Page;
+  selectedPage?: Page;
+  setSelectedPage: (pageId: string) => void;
+  onPanelPropertiesChange: () => void;
+  deletePage: (pageId?: string) => Promise<void>;
+  addQuestion: (questionType: PageQuestionType, pageId: string) => Promise<void>;
+}
+
+/**
+ * Editor preview component
+ */
+const EditorPreview = ({
+  htmlString,
+  width,
+  height,
+  scale,
+  page,
+  selectedPage,
+  setSelectedPage,
+  onPanelPropertiesChange,
+  deletePage,
+  addQuestion
+}: Props) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement>();
+  const [pageMenuOpen, setPageMenuOpen] = useState(false);
+  const [addQuestionDialogOpen, setAddQuestionDialogOpen] = useState(false);
+
+  if (!page.id) return null;
+
+  /**
+   * Set up event listener to recieve post message from iframe
+   */
+  useEffect(() => {
+    // TODO: In the future when we need more event handlers within the iframe we can use a switch statement to replace the handlePostMessageEventListener.
+    // This can direct us to the appropriate event listener based upon the type of the Event e.g. IframeClickEvent, IframeButtonClickEvent etc.
+    window.addEventListener(`message-${page?.id}`, handlePostMessageEventListener);
+
+    return () => window.removeEventListener(`message-${page?.id}`, handlePostMessageEventListener);
+  }, []);
+
+  /**
+   * Handles post messsage event from iframe
+   *
+   * @param event message event
+   */
+  const handlePostMessageEventListener = (event: any) => {
```

Comment:
As we are now using the CUstomEvent rather than PostMessage could this be renamed?

Reply:
Resolved